### ISSUE TITLE
[TypeScript] Invalid overload type when using `flow()`

### ISSUE THREAD

**Issue by leesiongchan** [mobxjs/mobx-state-tree#507]
```
const fetch = flow(function *(url: string, options: RequestInit = {}) {
...
});

const create = (url: string, data: {}) => {
  return fetch(url, {
    body: JSON.stringify(data),
    method: 'POST',
  });
};

// [ts] Expected 1 arguments, but got 2.
// const fetch: (a1: string) => Promise<any>
```

**Comment by mattiamanzati** [comment]
Unfortunately known TS issue of not having variadic types :/

**Comment by jiyinyiyong** [comment]
Got similar problem in `flow(f)` then I use an optional argument, fixed by removing `?`...

```js
loadM: flow(function* loadM(forceReload?: boolean) {
```

```
    TS2554: Expected 0 arguments, but got 1.
```

**Comment by mweststrate** [comment]
Closing until variadic types :'(

**Comment by benjick** [comment]
Any workaround?

**Comment by xaviergonz** [comment]
Reopening since now it has a PR

**Comment by khomyakov42** [comment]
> Any workaround?

You can declare "type" for arguments, example:
```typescript

type Filter = {
   id?: number
   owner?: number
}
function findOne(filter: Filter) {
   ...
}

```

**Comment by xaviergonz** [comment]
Closing, a fix was released with mst 3.9
